### ISSUE TITLE
Create schema for pipeline database

### ISSUE THREAD

**Issue by weaverba137** [desihub/desispec#320]
Based on @sbailey's slides from the OSU meeting this database should have this approximate structure:

* a brick table, with brickid or brickname as the primary key.
* a target table with targetid as the primary key.
* a truth table with arbitrary primary key, and a foreign key to the target table.
* a fiberassign/frame table with foreign keys to target and tile.
* a obslist table with a foreign key to the tile table.
* a tile table with tileid as the primary key.
* a zcat table with a foreign key to the target table.

This brings up a few questions that I can't find in my notes:

1. What is the relationship between target and truth? 1:1? 1:many? many:many?
2. In the fiberassign table, is every row a fiber?
3. The obslist table seems very similar to the frame table I developed for the raw data.  We need to capture the fact that a tile may be observed multiple times, possibly with different fiber assignments; in the frame table, the primary key was spectrograph arm plus exposure number.

**Comment by sbailey** [comment]
> 1. What is the relationship between target and truth? 1:1? 1:many? many:many?

1:1 row matched in the files, with TARGETID in both as a crosscheck on that.  i.e. the code could assert that `np.all(target['TARGETID'] == truth['TARGETID'])`.

> 2. In the fiberassign table, is every row a fiber?

yes

> 3. The obslist table seems very similar to the frame table I developed for the raw data. We need to capture the fact that a tile may be observed multiple times, possibly with different fiber assignments; in the frame table, the primary key was spectrograph arm plus exposure number.

> a fiberassign/frame table with foreign keys to target and tile.

Related information: https://desi.lbl.gov/trac/wiki/Pipeline/FormatsAndNumbering tried to define "tile" as the geometric location on the sky, while "pointing" was a specific pointing of the positioners to targets on that tile.  i.e. a single tile could have multiple pointings, especially during commissioning or other special tests.  We seem to have lost that distinction and may not currently support the ability to have multiple pointings per tile.  Bad.

"frame" is the data from a single arm of a single spectrograph from a single exposure, i.e. its primary key should be a combination of all 3 of those, not just arm+exposure number.

"fiberassign" applies to all 5000 fibers; i.e. it defines the pointing for a tile (despite the filename currently being "tile_{tileid}.fits").

An obslist row applies to the exposure as a whole, linking what tile/pointing was observed under what conditions.  A row of obslist corresponds to 30 frames (10 spectrographs x 30 arms).

observation:pointing is many:one
pointing:tile is many:one
pointing and fiberassign are basically the same thing (perhaps pointing is a realization of fiberassign)
exposure and observation are basically the same thing

**Comment by weaverba137** [comment]
> Related information: https://desi.lbl.gov/trac/wiki/Pipeline/FormatsAndNumbering tried to define "tile" as the geometric location on the sky, while "pointing" was a specific pointing of the positioners to targets on that tile. i.e. a single tile could have multiple pointings, especially during commissioning or other special tests. We seem to have lost that distinction and may not currently support the ability to have multiple pointings per tile. Bad.

Right, but by breaking tiles down into frames, we can retain this support by tying frames to pointings, then pointings to tiles.

> "frame" is the data from a single arm of a single spectrograph from a single exposure, i.e. its primary key should be a combination of all 3 of those, not just arm+exposure number.

Sorry, that's just me being sloppy with language. By "spectrograph" I actually meant arm+spectrograph number.  If you look in detail at the existing code, you'll see this is already implemented.

The rest all makes sense to me.
